lib/lexer.js

```diff
@@ -586,6 +586,21 @@ Lexer.prototype = {
     }
   },
 
+
+  /**
+   * Block code.
+   */
+
+  blockCode: function() {
+    var captures;
+    if (captures = /^-[ \t]*(\n|$)/.exec(this.input)) {
```

Comment:
As of now `-\t` will be parsed as source code - so if we want to avoid breaking changes it would probably best to stick to `/^-\n/` as proposed by @ForbesLindesay.